vignettes/germany-age-stratified-nowcasting.Rmd.orig

```diff
@@ -92,22 +88,28 @@ Here we explore a range of increasingly complex models using subject area knowle
 
 ## Shared reporting delay distribution 
 
-We first explore a relatively simple model that assumes that reporting delays are fixed across age groups and time. As this model is the default we simply call `epinowcast`. As we want to make use of `CmdStan`'s support for in-chain parallisation we first compile the default model with this enabled (because of this we also need to pass `threads_per_chain` to `epinowcast`).
+We first explore a relatively simple model that assumes that reporting delays are fixed across age groups and time. As this model is the default we simply call `epinowcast`. As we want to make use of `CmdStan`'s support for within-chain parallelisation we first compile the default model with this enabled (because of this we also need to pass `threads_per_chain` to `epinowcast`).
+
+```{r, include = FALSE, message = FALSE}
+multithread_model <- enw_model(threads = TRUE)
+```
 
-```{r, message = FALSE}
+```{r, message = FALSE, eval = FALSE}
 multithread_model <- enw_model(threads = TRUE)
 ```
 
-Note that here we use two chains each using `r  threads` threads as a demonstration but in general using 4 chains is recommended. Also note that here we have silenced fitting progress and potential warning messages but in general this should not be done.
+Note that here we use two chains each using `r  threads` threads as a demonstration but in general using 4 chains is recommended. Also note that warm-up and sampling iterations have been set below default values to reduce compute requirements but this may not be sufficient for many real world use cases. Finally, note that here we have silenced fitting progress and potential warning messages but in general this should not be done.
 
-```{r}
-options(mc.cores = 2)
+```{r simple-nowcast}
+fit <- enw_fit_opts(
+  save_warmup = FALSE, output_loglik = TRUE, pp = TRUE,
+  chains = 2, threads_per_chain = threads, 
+  iter_sampling = 500, iter_warmup = 250,
+  show_messages = FALSE, refresh = 0
+)
 nowcast <- epinowcast(pobs,
+  fit = fit,
   model = multithread_model,
```

Comment:
I am not sure if `fit` is ideal, as it represents the fitted model object in many other packages. Could `fit_opts` or `sampling` / `sampling_opts` be better?

Reply:
Good point. I wanted to steer clear of args with `_opts` in the title as it seemed like duplication (all arg are opts by definition here). I also wanted to steer clear of `sampling/sample` etc in case we implement something that doesn't do sampling as a back-end in the future (ha I know not close). 

Personally, I would vote we go with this for now and maybe evolve it in `0.2.0`? Now things are settling down a bit we can properly depreciate args using `lifecycle` so shouldn't lead to user pain.

Reply:
Going to resolve this but very keen to discuss more for `0.2.0` and onwards (need to not stare at this PR anymore!).